decidim/decidim
# Fixed a bug where admin can remove his admin rights

#### :tophat: What? Why?
There was a bug in Decidim in which an admin user could remove their admin rights from the admin panel. Also the same problem was in the system panel. For the organization admin panel there is now a permission change which checks if the current admin user tries to commit an "destroy" -action on themselves, they have no permission to do it, in fact it is impossible to do it in the panel since the lack of permission hides the button from the user. In the system panel a helper method was made which checks if the target of the action matches the current admin user and if it does, it also hides the button. There was no permission structure in system.
This problem is really frustrating especially in cases where there is only one admin user, and it is really hard to get your admin rights back after deletion.
#### :pushpin: Related Issues
- Fixes #9311

#### Testing
1. Login as an Admin user.
2. Go to the Admin panel
3. Click on "Participants", and then click "Admins"
4. Make sure there is only one Admin,which is yourself (the admin account you have logged in)
5. Click the "**x**" -button on the right side of the list, and in the dialog box confirm deleting
6. See error
### :camera: Screenshots
**Admin panel before bug**
![image](https://user-images.githubusercontent.com/110532525/210758009-5ddf3de3-8e37-410e-9732-fac0797d9620.png)
**Admin panel after bug**
![image](https://user-images.githubusercontent.com/110532525/210757860-e632d182-3ed9-4396-845f-913a6c2e03ca.png)
**System panel before bug**
![image](https://user-images.githubusercontent.com/110532525/210758784-1fd060cd-2f1b-488f-9f58-2f618a4e25da.png)

**System panel after bug**
![image](https://user-images.githubusercontent.com/110532525/210758542-bacf0888-e34e-4b20-898f-0b55b615dddf.png)


:hearts: Thank you!
